OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -87,6 +87,16 @@ void Frame::addMeshGeometry(const std::string& aGeometryFileName, const SimTK::V
     addGeometry(geom);
 }
 
+
+void Frame::attachGeometry(const OpenSim::Geometry& geom, const SimTK::Vec3 scale)
+{
+    SimTK::ClonePtr<Geometry> clone = SimTK::ClonePtr<Geometry>(geom);
+    clone->set_scale_factors(scale);
+    clone->setFrameName(getName());
+    addGeometry(clone.updRef());
```

Comment:
Here's the concern, Geometry is just base class, clone returns the correct concrete subclass. We want to make sure we preserve the actual concrete subclass, not sure you can do it with the above code, does that sound right?